### PR TITLE
Update handles

### PR DESCRIPTION
**Pull request by dholms**
Adds a method for updating handles. Pretty straightforward.

Only data integrity issue I can see is:
- the way we do mentions still encodes their old handle _as text_ in the post. So the link itself will still work (because it links to the DID), but the handle text will not change. I think this is probably the expected behavior

One thing to think through here that is not included in this PR is: how do username changes propagate to downstream systems?

Closes https://github.com/bluesky-social/atproto/issues/480
Prepares us for https://github.com/bluesky-social/atproto/issues/481

### CODE REVIEW

File: packages/pds/src/api/com/atproto/handle.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import * as handleLib from '@atproto/handle'
+import { Server } from '../../../lexicon'
+import AppContext from '../../../context'
+import { UserAlreadyExistsError } from '../../../services/actor'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.handle.resolve(async ({ params }) => {
```
Comment (dholms) [inline packages/pds/src/api/com/atproto/handle.ts:8]: nothing new here, just changed the file name to match the schema

File: packages/pds/tests/handles.test.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    )
+
+    const aliceFollows = followers.data.followers.filter((f) => f.did === alice)
+    expect(aliceFollows.length).toBe(1)
+    expect(aliceFollows[0].handle).toBe(newHandle)
+  })
+
+  it('does not allow taking a handle that already exists', async () => {
+    const attempt = agent.api.com.atproto.handle.update(
+      { handle: 'bob.test' },
```
Comment (devinivy) [inline packages/pds/tests/handles.test.ts:101]: Might suggest exercising the handle normalization here by futzing with casing:
```suggestion
      { handle: 'Bob.Test' },
```

File: packages/pds/src/services/actor/index.ts [hunk 3]
```diff
@@ -166,6 +166,23 @@ export class ActorService {
...
+    const res = await this.db.db
+      .updateTable('did_handle')
+      .set({ handle })
+      .where('did', '=', did)
+      .whereNotExists(
+        this.db.db
+          .selectFrom('did_handle')
+          .where('handle', '=', handle)
+          .selectAll(),
+      )
```
Comment (devinivy) [inline packages/pds/src/services/actor/index.ts:179]: We should be able to lean directly on the `handle_lower` index to ensure we don't get a duplicate, which I think would actually feel pretty nice, and prove that we don't strictly rely on trusted `handle` inputs to this method.  That would involve switching this to an `onConflict((oc) => oc.doNothing())`.

**Review by devinivy — APPROVED** [review]
Good looks!

File: packages/pds/tests/handles.test.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/tests/handles.test.ts:101]: yup im game 👌

File: packages/pds/src/services/actor/index.ts (hunk 3 above)
Comment (dholms) [reply packages/pds/src/services/actor/index.ts:179]: I would prefer that too & it's what I originally tried, but there's no `onConflict` for `update` :/

Any other ideas?

Comment (devinivy) [reply packages/pds/src/services/actor/index.ts:179]: Ah dang, good call!  My only other idea is to handle the constraint error that comes out of the database, but that's a little funky to do across sqlite and pg.  I think as-is works well particularly with a test to back it up 👍

Comment (dholms) [reply packages/pds/src/services/actor/index.ts:179]: cool cool 👍 

got that test in & it works fine

**Review by devinivy — APPROVED** [review]

**Pull request merged**